tactics/hints.ml

```diff
@@ -1704,7 +1704,7 @@ let pr_hint_db_env env sigma db =
     else str" (modes " ++ prlist_with_sep pr_comma pr_mode l ++ str")"
   in
   let content =
-    let fold head modes hintlist accu =
+    let fold accu (head, modes, hintlist) =
```

Comment:
```suggestion
    let pr_one (head, modes, hintlist) =
```
then instead of List.fold_left use Pp.prlist
(might cause the printing order to be reversed in which case just flip the comparator given to the sort function)